input/kafkamdm/kafkamdm.go

```diff
@@ -256,13 +248,11 @@ func (k *KafkaMdm) tryGetOffset(topic string, partition int32, offset int64, att
 	return val, err
 }
 
-// this will continually consume from the topic until k.stopConsuming is triggered.
+// consumePartition consumes from the topic until k.shutdown is triggered.
 func (k *KafkaMdm) consumePartition(topic string, partition int32, currentOffset int64) {
 	defer k.wg.Done()
 
-	partitionOffsetMetric := partitionOffset[partition]
-	partitionLogSizeMetric := partitionLogSize[partition]
-	partitionLagMetric := partitionLag[partition]
+	kafkaStats := kafkaStats[partition]
```

Comment:
detail: this could be initialized further down on `:274` so we can save the allocation if there's an error in getting the offset